src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -111,6 +127,54 @@ internal static int GetMemberArityIncludingExtension(this Symbol member)
             return member.GetMemberArity();
         }
 
+        internal static ImmutableArray<TypeParameterSymbol> GetTypeParametersIncludingExtension<TMember>(this TMember member) where TMember : Symbol
+        {
+            if (member is MethodSymbol method)
+            {
+                return method.GetIsNewExtensionMember()
+                    ? method.ContainingType.TypeParameters.Concat(method.TypeParameters)
+                    : method.ConstructedFrom.TypeParameters;
+            }
+
+            if (member is PropertySymbol property)
+            {
+                Debug.Assert(property.GetIsNewExtensionMember());
```

Comment:
>Debug.Assert(property.GetIsNewExtensionMember()); [](http://example.com/codeflow?start=16&length=49)

It feels inconsistent that properties and methods get different treatment. For methods we check this condition, but for properties we assume it is true.